src/Common/CustomAttributes/CmdletPreviewAttribute.cs

```diff
@@ -25,19 +26,36 @@ public class CmdletPreviewAttribute : System.Attribute
     {
         public string _message;
 
+        public DateTime EstimatedGaDate { get; }
+
+        public bool IsEstimatedGaDateSet { get; } = false;
+
         public CmdletPreviewAttribute()
         {
             this._message = Resources.PreviewCmdletMessage;
         }
 
         public CmdletPreviewAttribute(string message)
         {
-            this._message = message;
+            this._message = string.IsNullOrEmpty(message) ? Resources.PreviewCmdletMessage : message;
+        }
+
+        public CmdletPreviewAttribute(string message, string estimatedDateOfGa) : this(message)
```

Comment:
Do we already have cases of GA date in azure-powershell repo?

Reply:
Searched whole repo, no such case is found.